CSCoreAnnotation/robot.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+#
+
+
+import threading
+import time
+
+import cv2
+import numpy as np
+
+import wpilib
+from cscore import CameraServer
+
+
+class MyRobot(wpilib.TimedRobot):
+    """
+    This is a sample program showing how to overlay a red rectangle on a camera feed.
+    The rectangle's position is controlled from robot code during teleop.
+    This requires at least one USB camera connected to the robot.
+    Use a dashboard client to view the camera stream. Two streams are available:
+    - "Annotated Stream": The camera feed with the red rectangle overlay.
+    - "<your camera name>": The raw camera feed without any annotations.
+    """
+
+    def robotInit(self):
+        """Robot initialization function"""
+
+        CameraServer.enableLogging()
+
+        # State variables - this will be the center position of the rectangle, in units of pixels.
+        self.rectX = 0
+        self.rectY = 0
+
+        # Start reading data from the camera, and serving the raw stream
+        self.camera = CameraServer.startAutomaticCapture()
+        self.camera.setResolution(640, 480)
+
+        # Record a reference to the video source (which should be the first camera connected):
+        self.source = self.camera.enumerateSources()
+
+        # Set up a second server to provide an annotated stream
+        self.cvSink = CameraServer.getVideo()
+        self.outputStream = CameraServer.putVideo("Annotated Stream", 640, 480)
+
+        # Allocate memory for creating the annotated image.
+        self.img = np.zeros((480, 640, 3), dtype=np.uint8)
+
+        # Start the background thread for image processing
+        self.processingThread = threading.Thread(target=self.processImages, daemon=True)

```

Comment:
Threading in Python isn't a good idea on a robot whilst the GIL is still a thing, especially for a heavy workload like image processing. That's why we have a `wpilib.CameraServer.launch` that spawns a subprocess.

FWIW the majority of our vision examples live in https://github.com/robotpy/mostrobotpy/tree/main/examples/cscore. I do see the merit of having a full robot project example in here though, and I think we do have a couple of those.

Reply:
See documentation at https://docs.wpilib.org/en/stable/docs/software/vision-processing/roborio/using-the-cameraserver-on-the-roborio.html

Reply:
I'm great at reading docs! Yay! 

Yea, fair, threading likely won't work well.

Since data passing is happening, maybe subprocesses with pipes? It's a bit more complex but probably manageable. Would need some tweaks to that doc page too. Alternatively, accepting a lower framerate at the output and injecting the image annotation inline with the main periodic loop (again, not great, but at least won't starve main robot code of any execution)

This wasn't much effort, honestly like just 15 minutes of pre-work while I was thinking about how to tweak the original doc page with an outdated example. I'm happy to put a lid on this one (esp. till 2027 where maybe the advice changes?), and just address the actual docs issue?